Huh, I guess adding `count` to your type changes the diagnostics you get.

CC @rudkx @xedin, any thoughts on what the behavior should be?


```
******************** TEST 'Swift(macosx-x86_64) :: Constraints/diagnostics.swift' FAILED ********************
17:25:19 Script:
17:25:19 --
17:25:19 /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/bin/swiftc -frontend -target x86_64-apple-macosx10.9  -module-cache-path '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/swift-test-results/x86_64-apple-macosx10.9/clang-module-cache' -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -swift-version 3  -typecheck -verify -disable-objc-attr-requires-foundation-module /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/Constraints/diagnostics.swift
17:25:19 --
17:25:19 Exit Code: 1
17:25:19 
17:25:19 Command Output (stderr):
17:25:19 --
17:25:19 
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/Constraints/diagnostics.swift:164:43: error: incorrect message found
17:25:19   if s.count() == 0 {} // expected-error{{cannot call value of non-function type 'Int'}}{{13-15=}}
17:25:19                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
17:25:19                                           cannot invoke 'count' with no arguments
17:25:19 
17:25:19 --
17:25:19 
```

--- a/test/Constraints/diagnostics.swift
+++ b/test/Constraints/diagnostics.swift
@@ -161,7 +161,8 @@ func rdar20142523() {
 // <rdar://problem/21080030> Bad diagnostic for invalid method call in boolean expression: (_, ExpressibleByIntegerLiteral)' is not convertible to 'ExpressibleByIntegerLiteral
 func rdar21080030() {
   var s = "Hello"
-  if s.count() == 0 {} // expected-error{{cannot call value of non-function type 'Int'}}{{13-15=}}
+  // SR-7599: This should be `cannot_call_non_function_value`
+  if s.count() == 0 {} // expected-error{{cannot invoke 'count' with no arguments}}
 }
 
 // <rdar://problem/21248136> QoI: problem with return type inference mis-diagnosed as invalid arguments

